File: capture-server/src/event-preview.cpp
#include "event-preview.h"
#include "gateway.h"
#include "event.h"
#include <json/json.h>

EventPreview::EventPreview(Json::Value &model) : Base(model)
{
}

void EventPreview::report()
{
    Base::report();
}

std::string EventPreview::dtEvent() const
{
    return m_model["dtEvent"].asString();
}

void EventPreview::setDtEvent(const std::string &value)
{
    m_model["dtEvent"] = value;
}

std::string EventPreview::level() const
{
    return m_model["level"].asString();
}

void EventPreview::setLevel(const std::string &value)
{
    m_model["level"] = value;
}

std::string EventPreview::program() const
{
    return m_model["program"].asString();
}

void EventPreview::setProgram(const std::string &value)
{
    m_model["program"] = value;
}

std::string EventPreview::sport() const
{
    return m_model["sport"].asString();
}

void EventPreview::setSport(const std::string &value)
{
    m_model["sport"] = value;
}

std::string EventPreview::status() const
{
    return m_model["status"].asString();
}

void EventPreview::setStatus(const std::string &value)
{
    m_model["status"] = value;
}

std::string EventPreview::title() const
{
    return m_model["title"].asString();
}

void EventPreview::setTitle(const std::string &value)
{
    m_model["title"] = value;
}

std::string EventPreview::cityAddress() const
{
    return m_model["detail"]["cityAddress"].asString();
}

void EventPreview::setCityAddress(const std::string &value)
{
    m_model["detail"]["cityAddress"] = value;
}

std::string EventPreview::streetAddress() const
{
    return m_model["detail"]["streetAddress"].asString();
}

void EventPreview::setStreetAddress(const std::string &value)
{
    m_model["detail"]["streetAddress"] = value;
}

std::string EventPreview::detailType() const
{
    return m_model["detail"]["type"].asString();
}

void EventPreview::setDetailType(const std::string &value)
{
    m_model["detail"]["type"] = value;
}

std::string EventPreview::venueLocation() const
{
    return m_model["venue"]["location"].asString();
}

void EventPreview::setVenueLocation(const std::string &value)
{
    m_model["venue"]["location"] = value;
}
 
int EventPreview::year() const
{
    return m_model["year"].asInt();
}

void EventPreview::setYear(int value)
{
    m_model["year"] = value;
}

int EventPreview::time() const
{
    return m_model["time"].asInt();
}

void EventPreview::setTime(int value)
{
    m_model["time"] = value;
}

std::string EventPreview::eventType() const
{
    return m_model["type"].asString();
}

void EventPreview::setEventType(const std::string &value)
{
    m_model["type"] = value;
}

std::string EventPreview::countdown() const
{
    return m_model["countdown"].asString();
}

void EventPreview::setCountdown(const std::string &value)
{
    m_model["countdown"] = value;
}

std::vector<EventDevice> &EventPreview::activeDevices()
{
    return m_activeDevice;
}

const std::vector<EventDevice> &EventPreview::activeDevices() const
{
    return m_activeDevice;
}

void EventPreview::setActiveDevices(const std::vector<EventDevice> &activeDevices)
{
    // Clear the existing active devices in the model
    m_model["activeDevices"] = Json::arrayValue;
    // Check if the provided vector is empty
    if (activeDevices.empty())
    {
        // Throw an exception indicating that no active devices were provided
        spdlog::warn("No active devices for the event exist..");
    }
    else
        for (const auto &device : activeDevices)
        {
            Json::Value jsonDevice;
            jsonDevice["deviceId"] = device.deviceId();
            jsonDevice["eventId"] = device.eventId();
            jsonDevice["location"] = device.location();
            jsonDevice["pin"] = device.pin();
            jsonDevice["userId"] = device.userId();
            m_model["activeDevices"].append(jsonDevice);
        }
}

File: capture-server-ui/src/app/sport-pip-capture/components/event/event-preview/views/event-preview.ts
import { Collection } from "app/blocks/collection";
import { Views } from "app/sport-pip-capture/models/capture-interface";
import { Range } from "app/blocks/collection";
export class RangeEventPreviewView implements Views.Datasource {

  private _id: number;
  public get id(): number {
    return this._id;
  }
  public set id(v: number) {
    this._id = v;
  }


  private _eventPreview: Range<EventPreview>;
  public get eventPreview(): Range<EventPreview> {
    if (!this._eventPreview) {
      this._eventPreview = new Range<EventPreview>();
    }
    return this._eventPreview;
  }
  public set eventPreview(v: Range<EventPreview>) {
    this._eventPreview = v;
  }

}
export class EventPreview implements Views.Datasource {
  private _id: number;
  private _dtEvent: string;
  private _level: string;
  private _program: string;
  private _sport: string;
  private _status: string;
  private _title: string;
  private _detail: {
    cityAddress: string;
    streetAdress: string;
    type: string;
  };
  private _year: number;
  private _venue: {
    location: string;
  };
  private _time: number;
  private _type: string;
  private _countdown: string;
  private _activeDevice: Collection<any>;

  // Getters
  get id(): number {
    return this._id;
  }

  get dtEvent(): string {
    return this._dtEvent;
  }

  get level(): string {
    return this._level;
  }

  get program(): string {
    return this._program;
  }

  get sport(): string {
    return this._sport;
  }

  get status(): string {
    return this._status;
  }

  get title(): string {
    return this._title;
  }

  get detail(): { cityAddress: string; streetAdress: string; type: string } {
    return this._detail;
  }

  get year(): number {
    return this._year;
  }

  get venue(): { location: string } {
    return this._venue;
  }

  get time(): number {
    return this._time;
  }

  get type(): string {
    return this._type;
  }

  get countdown(): string {
    return this._countdown;
  }

  // get activeDevice(): Collection<ActiveDeviceView> {
  //   if (!this._activeDevice) {
  //     this._activeDevice = new Collection<ActiveDeviceView>();
  //   }
  //   return this._activeDevice;
  // }

  // Setters
  set id(value: number) {
    this._id = value;
  }

  set dtEvent(value: string) {
    this._dtEvent = value;
  }

  set level(value: string) {
    this._level = value;
  }

  set program(value: string) {
    this._program = value;
  }

  set sport(value: string) {
    this._sport = value;
  }

  set status(value: string) {
    this._status = value;
  }

  set title(value: string) {
    this._title = value;
  }

  set detail(value: {
    cityAddress: string;
    streetAdress: string;
    type: string;
  }) {
    this._detail = value;
  }

  set year(value: number) {
    this._year = value;
  }

  set venue(value: { location: string }) {
    this._venue = value;
  }

  set time(value: number) {
    this._time = value;
  }

  set type(value: string) {
    this._type = value;
  }

  set countdown(value: string) {
    this._countdown = value;
  }


  private _previewActiveDevice: Range<ActiveDeviceView>;
  public get previewActiveDevice(): Range<ActiveDeviceView> {
    if (!this._previewActiveDevice) {
      this._previewActiveDevice = new Range<ActiveDeviceView>()
    }
    return this._previewActiveDevice;
  }
  public set previewActiveDevice(v: Range<ActiveDeviceView>) {
    this._previewActiveDevice = v;
  }

}

export class ActiveDeviceView {

  private _network: string;
  public get network(): string {
    return this._network;
  }
  public set network(v: string) {
    this._network = v;
  }

  private _deviceType: string;
  public get deviceType(): string {
    return this._deviceType;
  }
  public set deviceType(v: string) {
    this._deviceType = v;
  }

  private _deviceId: string;
  public get deviceId(): string {
    return this._deviceId;
  }
  public set deviceId(v: string) {
    this._deviceId = v;
  }

  private _location: string;
  public get location(): string {
    return this._location;
  }
  public set location(v: string) {
    this._location = v;
  }

  private _user: string;
  public get user(): string {
    return this._user;
  }
  public set user(v: string) {
    this._user = v;
  }
}

export class VenueView {
  private _location: string;
  public get location(): string {
    return this._location;
  }
  public set location(v: string) {
    this._location = v;
  }
}
